spec/add_gist_spec.rb

```diff
@@ -0,0 +1,38 @@
+require 'add_gist'
+
+describe AddGist do
+
+  describe '.read_files' do
+    context 'given no path' do
+      it 'returns files on current directory' do
+        expect(AddGist.read_files('add_gist.rb')). to have_key('add_gist.rb')
+      end
+    end
+
+    context 'given unexisting path' do
+      it 'returns error' do
+        expect(AddGist.read_files('xyz')).to raise_error(Errno::ENOENT, 'The path xyz does not exist.')
+      end
+    end
+
+    context 'given empty directory path' do
+      it 'returns no files' do
+        expect(AddGist.read_files('empty')).to be_empty
+      end
+    end
+
+    context 'given existing directory path' do
+      it 'returns all files' do
+        hash = AddGist.read_files('prueba')
```

Comment:
I think this should be mocked, because if we run the tests in a separate environment (e.g. circle ci) without having the `prueba` directory this test will fail.